File: 2640-maximum-number-of-integers-to-choose-from-a-range-i/2640-maximum-number-of-integers-to-choose-from-a-range-i.java
class Solution {
    public int maxCount(int[] banned, int n, int maxSum) {
        HashSet<Integer>set = new HashSet<>();
        for(int num : banned){
            set.add(num);
        }
        int currentSum = 0;
        int count = 0;
        for(int i=1;i<=n;i++){
            currentSum += i;
            if(set.contains(i) || (currentSum > maxSum)){
                currentSum -= i;
            }else{
                count++;
            }

        }
        return count;
        
    }
}

File: 2640-maximum-number-of-integers-to-choose-from-a-range-i/2640-maximum-number-of-integers-to-choose-from-a-range-i.js
/**
 * @param {number[]} banned
 * @param {number} n
 * @param {number} maxSum
 * @return {number}
 */
var maxCount = function(banned, n, maxSum) {
    let currentSum = 0;
    let count = 0;
    for(let i=1;i<=n;i++){
        currentSum += i;
        if(banned.includes(i) || currentSum>maxSum){
            currentSum -= i;
            
        }else{
            count++;
        }
    }
    return count;
    
};